src/app/Http/Controllers/CrudFeatures/SaveActions.php

```diff
@@ -12,28 +12,41 @@
     public function getSaveAction()
     {
         $saveAction = session('save_action', config('backpack.crud.default_save_action', 'save_and_back'));
-        $saveOptions = [];
-        $saveCurrent = [
-            'value' => $saveAction,
-            'label' => $this->getSaveActionButtonName($saveAction),
+
+        // Permissions and their related actions.
+        $permissions = [
+            'list'   => 'save_and_back',
+            'update' => 'save_and_edit',
+            'create' => 'save_and_new',
         ];
 
-        switch ($saveAction) {
-            case 'save_and_edit':
-                $saveOptions['save_and_back'] = $this->getSaveActionButtonName('save_and_back');
-                $saveOptions['save_and_new'] = $this->getSaveActionButtonName('save_and_new');
-                break;
-            case 'save_and_new':
-                $saveOptions['save_and_back'] = $this->getSaveActionButtonName('save_and_back');
-                $saveOptions['save_and_edit'] = $this->getSaveActionButtonName('save_and_edit');
-                break;
-            case 'save_and_back':
-            default:
-                $saveOptions['save_and_edit'] = $this->getSaveActionButtonName('save_and_edit');
-                $saveOptions['save_and_new'] = $this->getSaveActionButtonName('save_and_new');
-                break;
+        $saveOptions = collect($permissions)
```

Comment:
Personally I'd life this out so this read:

```
$saveOptions = $this->getAllowedSaveOptions($permissions);
```

But OTOH it's not too bad as it is.

Reply:
If it was used multiple places I'd agree.
The `->toArray()` was changed to `->all()`, but it *is* needed since `Collection` doesn't support `key()`, thus you'd have to do `$collection->keys->first()` to accomplish the same.